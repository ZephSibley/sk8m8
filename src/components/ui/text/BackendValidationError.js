import React from 'react';
import FormHelperText from '@material-ui/core/FormHelperText';

const BackendValidationError = ({ resp }) => {
    
    return (
        <FormHelperText
            component={'span'}
            error={true}
        >
            {resp}
        </FormHelperText>
    )
}

export default BackendValidationError;