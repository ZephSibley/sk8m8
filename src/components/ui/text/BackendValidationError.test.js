import React from 'react';
import renderer from 'react-test-renderer';
import BackendValidationError from './BackendValidationError';

test('Rendering a string', () => {
    const component = renderer.create(
        <BackendValidationError resp='this is a string' />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})

test('Rendering an array', () => {
    const component = renderer.create(
        <BackendValidationError resp={['rendering', 'an', 'array']} />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})
