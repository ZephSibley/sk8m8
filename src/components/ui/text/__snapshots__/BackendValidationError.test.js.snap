// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering a string 1`] = `
<pre
  className="MuiFormHelperText-root Mui-error"
>
  this is a string
</pre>
`;

exports[`Rendering an array 1`] = `
<pre
  className="MuiFormHelperText-root Mui-error"
>
  rendering
  an
  array
</pre>
`;
